python/cugraph/structure/graph_implementation/simpleGraph.py

```diff
@@ -0,0 +1,838 @@
+            df['out_degree'] : cudf.Series
+                The out-degree of the vertex.
+        Examples
+        --------
+        >>> M = cudf.read_csv('datasets/karate.csv', delimiter=' ',
+        >>>                   dtype=['int32', 'int32', 'float32'], header=None)
+        >>> G = cugraph.Graph()
+        >>> G.from_cudf_edgelist(M, '0', '1')
+        >>> df = G.degrees([0,9,12])
+        """
+        (
+            vertex_col,
+            in_degree_col,
+            out_degree_col,
+        ) = graph_primtypes_wrapper._degrees(self)
+
+        df = cudf.DataFrame()
+        df["vertex"] = vertex_col
+        df["in_degree"] = in_degree_col
+        df["out_degree"] = out_degree_col
+
+        if self.properties.renumbered is True:
+            df = self.renumber_map.unrenumber(df, "vertex")
+
+        if vertex_subset is not None:
+            df = df[df['vertex'].isin(vertex_subset)]
+
+        return df
+
+    def _degree(self, vertex_subset, x=0):
+        vertex_col, degree_col = graph_primtypes_wrapper._degree(self, x)
+        df = cudf.DataFrame()
+        df["vertex"] = vertex_col
+        df["degree"] = degree_col
+
+        if self.properties.renumbered is True:
+            df = self.renumber_map.unrenumber(df, "vertex")
+
+        if vertex_subset is not None:
+            df = df[df['vertex'].isin(vertex_subset)]
+
+        return df
+
+    def to_directed(self, DiG):
+        """
+        Return a directed representation of the graph.
+        This function sets the type of graph as DiGraph() and returns the
+        directed view.
+        Returns
+        -------
+        G : DiGraph
+            A directed graph with the same nodes, and each edge (u,v,weights)
+            replaced by two directed edges (u,v,weights) and (v,u,weights).
+        Examples
+        --------
+        >>> M = cudf.read_csv('datasets/karate.csv', delimiter=' ',
+        >>>                   dtype=['int32', 'int32', 'float32'], header=None)
+        >>> G = cugraph.Graph()
+        >>> G.from_cudf_edgelist(M, '0', '1')
+        >>> DiG = G.to_directed()
+        """
+        DiG.properties.renumbered = self.properties.renumbered
+        DiG.renumber_map = self.renumber_map
+        DiG.edgelist = self.edgelist
+        DiG.adjlist = self.adjlist
+        DiG.transposedadjlist = self.transposedadjlist
+
+    def to_undirected(self, G):
+        """
+        Return an undirected copy of the graph.
+        Returns
+        -------
+        G : Graph
+            A undirected graph with the same nodes, and each directed edge
+            (u,v,weights) replaced by an undirected edge (u,v,weights).
+        Examples
+        --------
+        >>> M = cudf.read_csv('datasets/karate.csv', delimiter=' ',
+        >>>                   dtype=['int32', 'int32', 'float32'], header=None)
+        >>> DiG = cugraph.DiGraph()
+        >>> DiG.from_cudf_edgelist(M, '0', '1')
+        >>> G = DiG.to_undirected()
+        """
```

Comment:
same feedback for this method: docstring doesn't match code, should probably assert the correct incoming type.

Reply:
doc updated